moshi/src/main/java/com/squareup/moshi/ClassJsonAdapter.java

```diff
@@ -141,13 +144,19 @@ private boolean includeField(boolean platformType, int modifiers) {
     try {
       reader.beginObject();
       while (reader.hasNext()) {
-        String name = reader.nextName();
-        FieldBinding<?> fieldBinding = fieldsMap.get(name);
-        if (fieldBinding != null) {
-          fieldBinding.read(reader, result);
+        int index = reader.selectName(selection);
+        FieldBinding<?> fieldBinding;
+        if (index != -1) {
+          fieldBinding = fieldsArray[index];
         } else {
-          reader.skipValue();
+          String name = reader.nextName();
+          fieldBinding = fieldsMap.get(name);
```

Comment:
Is this code path necessary? When would this result in a hit when the selection did not?